### ISSUE TITLE
Weapons stay cvar

### ISSUE THREAD

**Issue by Dracant** [jdolan/quetoo#412]
Requesting a cvar to force weapons stay so you can longer pick up ammo from a weapon after you have it..  Being able to pick up weapons over and over again and resupplying ammo, allows for abuse in defense style play, particularly ctf maps.

**Comment by kaadmy** [comment]
Not sure exactly how this works, does this leave the weapon available after picking it up for everyone or just you?
I've never played with this enabled, so I'm not sure how it works.

**Comment by Dracant** [comment]
All weapons are visible all the time. When a player walks over a weapon he has not yet picked up, he will acquire the weapon and some ammo. 
If the players walks over the weapon again, nothing happens.

**Comment by jdolan** [comment]
We *do* respawn weapons every 5 seconds. Is that really not enough for highly populated servers?

**Comment by WickedShell** [comment]
@jdolan: I think the complaint is partly that it means you get ammo every 5
seconds without leaving the area (ammo at half rate, but still, it is more
ammo?)

**Comment by Dracant** [comment]
This is exactly right. Someone can stand in one spot and stock up on ammo
on each weapon respawn.  Weapons stay is a common setting in Q2 for sure
and I think q3 as well.  Weapons stay forces those on defensive positions
to leave to stock up on ammo.

Weapons stay is especially important in ctf.

Thanks!

**Comment by kaadmy** [comment]
Just put this snippet of code in https://github.com/jdolan/quetoo/blob/master/src/game/default/g_weapon.c#L71

```
	if (other->client->locals.inventory[ent->locals.item->index] && g_weapons_stay->integer) {
		return false;
	}
```

That should implement weapons stay from what I understand, have to add the CVar separately.

**Comment by kaadmy** [comment]
Agh, comment is too close to the close button, sorry.

And this'll need to have much faster/instant respawn, too.. maybe half second or something?

**Comment by Dracant** [comment]
In q2 and q3, the weapon never disappears at all when weapons stay is enabled. I guess an instant respawn would work?

**Comment by Paril** [comment]
Instant respawn would mean you'd get infinite ammo. The way it works in Q2/Q3 is it doesn't despawn at all, it just picks up the weapon and doesn't free the entity and you can't pick it up again if you have the ammo.

I agree the weapon respawns are too quick. It's nice to always be able to find a weapon, but at the same time weapon's stay is 100% more effective at preventing "empty weapon syndrome" than having weapons respawn every 5 secs which is basically infinite ammo

**Comment by Paril** [comment]
Implemented in 10ba740

**Comment by Dracant** [comment]
Works perfectly!!

Nice work!!
